underfs/swift/src/main/java/tachyon/underfs/swift/direct/SwiftDirectUnderFileSystem.java

```diff
@@ -0,0 +1,400 @@
+import java.util.Set;
+
+import org.javaswift.joss.client.factory.AccountConfig;
+import org.javaswift.joss.client.factory.AccountFactory;
+import org.javaswift.joss.client.factory.AuthenticationMethod;
+import org.javaswift.joss.model.Access;
+import org.javaswift.joss.model.Account;
+import org.javaswift.joss.model.Container;
+import org.javaswift.joss.model.Directory;
+import org.javaswift.joss.model.DirectoryOrObject;
+import org.javaswift.joss.model.StoredObject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.map.SerializationConfig;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.underfs.UnderFileSystem;
+import tachyon.underfs.swift.direct.http.SwiftDirectClient;
+
+/**
+ * Under file system implementation for OpenStack Swift based on
+ * the JOSS library.
+ */
+public class SwiftDirectUnderFileSystem extends UnderFileSystem {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  /** Suffix for an empty file to flag it as a directory */
+  private static final String FOLDER_SUFFIX = "_$folder$";
+  /** Value used to indicate nested structure in Swift */
+  private static final String PATH_SEPARATOR = "/";
+
+  /** Swift account */
+  private final Account mAccount;
+  /** Container name of user's configured Tachyon container */
+  private final String mContainerName;
+  /** Prefix of the container, for example swift://my-container-name/ */
+  private final String mContainerPrefix;
+  /** JOSS access object */
+  private final Access mAccess;
+
+  public SwiftDirectUnderFileSystem(String containerName,
+      TachyonConf tachyonConf) throws Exception {
+    super(tachyonConf);
+    LOG.debug("Constructor init: {}", containerName);
+    AccountConfig config = new AccountConfig();
+    config.setUsername(tachyonConf.get(Constants.SWIFT_USER_KEY));
+    config.setTenantName(tachyonConf.get(Constants.SWIFT_TENANT_KEY));
+    config.setPassword(tachyonConf.get(Constants.SWIFT_API_KEY));
+    config.setAuthUrl(tachyonConf.get(Constants.SWIFT_AUTH_URL_KEY));
+    String authMethod = tachyonConf.get(Constants.SWIFT_AUTH_METHOD_KEY);
+    if (authMethod != null && authMethod.equals("keystone")) {
+      config.setAuthenticationMethod(AuthenticationMethod.KEYSTONE);
+    } else {
+      config.setAuthenticationMethod(AuthenticationMethod.TEMPAUTH);
+    }
+
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.configure(SerializationConfig.Feature.WRAP_ROOT_VALUE, true);
+    mContainerName = containerName;
+    mAccount = new AccountFactory(config).createAccount();
+    mAccess = mAccount.authenticate();
+    LOG.debug("Check if container: {} exists ", containerName);
+    Container containerObj = mAccount.getContainer(containerName);
+    if (!containerObj.exists()) {
+      containerObj.create();
+    }
+    mContainerPrefix = Constants.HEADER_SWIFT + mContainerName
```

Comment:
NIT: it doesn't look like we are hitting the 100-character line-width limit yet.